src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,278 @@
+import kitchenpos.infra.DefaultPurgomalumClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static kitchenpos.Fixtures.createMenu_두마리_치킨;
+import static kitchenpos.Fixtures.메뉴그룹_생성;
+import static kitchenpos.Fixtures.메뉴_상품_생성;
+import static kitchenpos.Fixtures.상품_생성;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("메뉴 서비스 테스트")
+class MenuServiceTest {
+
+    @Mock
+    private MenuRepository menuRepository;
+
+    @Mock
+    private MenuGroupRepository menuGroupRepository;
+
+    @Mock
+    private ProductRepository productRepository;
+
+    @Mock
+    private DefaultPurgomalumClient defaultPurgomalumClient;
+
+    @InjectMocks
+    private MenuService menuService;
+
+    private MenuGroup menuGroup;
+
+    private Product product;
+
+    @BeforeEach
+    void setUp() {
+        product = 상품_생성("후라이드", 16_000L);
+        menuGroup = 메뉴그룹_생성("치킨 세트");
+    }
+
+    @Test
+    @DisplayName("메뉴를 등록 할 수 있다")
+    void saveMenu() {
+        // given
+        given(menuGroupRepository.findById(any())).willReturn(java.util.Optional.of(menuGroup));
+        given(productRepository.findById(any())).willReturn(java.util.Optional.of(product));
+        given(productRepository.findAllByIdIn(any())).willReturn(List.of(product));
+        given(menuRepository.save(any())).willReturn(Fixtures.메뉴_생성_두마리_치킨());
+
+        // when
+        Menu 응답_결과 = menuService.create(Fixtures.메뉴_생성_두마리_치킨());
+
+        // then
+        assertThat(응답_결과).isNotNull();
+    }
+
+    @Test
+    @DisplayName("메뉴는 메뉴 그룹에 속해야 한다.")
+    void requireMenuGroup() {
+        // given
+        Menu 요청_객체 = createMenu_두마리_치킨(BigDecimal.valueOf(16_000L), null, menuGroup, List.of(메뉴_상품_생성(product, 1)), "저녁 안주", true);
+
+        // when & then
+        assertThatThrownBy(() -> menuService.create(요청_객체))
+                .isInstanceOf(NoSuchElementException.class);
+    }
+
+    @Test
+    @DisplayName("메뉴는 적어도 하나의 메뉴 항목을 포함해야 한다.")
+    void requireMenuProduct() {
+        // given
+        given(menuGroupRepository.findById(any())).willReturn(java.util.Optional.of(menuGroup));
+        Menu 요청_객체 = createMenu_두마리_치킨(BigDecimal.valueOf(16_000L), menuGroup.getId(), menuGroup, null, "저녁 안주", true);
```

Comment:
```suggestion
        Menu 메뉴가_없는_요청 = createMenu_두마리_치킨(BigDecimal.valueOf(16_000L), menuGroup.getId(), menuGroup, null, "저녁 안주", true);
```
요청 객체라고 정의하기 보단 어떤 요청인지 나타내주는것도 코드를 읽는데 큰 도움이 될것 같아요 😄